modules/rotating-bound-service-keys.adoc

```diff
@@ -0,0 +1,404 @@
+
+* To reduce the risk of authentication failures, ensure that you understand and prepare for the time-sensitive steps.
+
+* During this process, you must refresh all service accounts and restart all pods on the cluster.
+These actions are disruptive to workloads.
+To mitigate this impact, you can temporarily halt these services and then redeploy them when the cluster is ready.
+====
+
+.Prerequisites
+
+* You have access to the {oc-first} as a user with the `cluster-admin` role.
+ifdef::aws[* You have access to the AWS CLI (`aws`) as a user with privileges to manage Identity and Access Management (IAM) and workload identity configurations.]
+ifdef::gcp[* You have access to the Google Cloud CLI (`gcloud`) as a user with privileges to manage Identity and Access Management (IAM) and workload identity configurations.]
+ifdef::azure[* You have access to the Azure CLI (`az`) as a user with privileges to manage Identity and Access Management (IAM) and workload identity configurations.]
+* You have configured the `ccoctl` utility.
+* Your cluster is in a stable state.
+You can confirm that the cluster is stable by running the following command:
++
+[source,terminal]
+----
+$ oc adm wait-for-stable-cluster --minimum-stable-period=5s
+----
+
+.Procedure
+
+. Configure the following environment variables:
++
+[source,text]
+----
+CLUSTER_ID=$(oc get infrastructures cluster -o jsonpath='{.status.infrastructureName}')
+CLUSTER_NAME=${CLUSTER_ID:0:-6}
+TEMPDIR=${CLUSTER_NAME}-$(date +%Y%m%d%H%M%S)
+ifdef::gcp[]
+CURRENT_ISSUER=$(oc get authentication cluster -o jsonpath='{.spec.serviceAccountIssuer}')
+GCP_BUCKET=$(echo ${CURRENT_ISSUER} | cut -d "/" -f4)
+endif::gcp[]
+ifdef::azure[]
+CURRENT_ISSUER=$(oc get authentication cluster -o jsonpath='{.spec.serviceAccountIssuer}')
+AZURE_STORAGE_ACCOUNT=$(echo ${CURRENT_ISSUER} | cut -d "/" -f3 | cut -d "." -f1)
+AZURE_STORAGE_CONTAINER=$(echo ${CURRENT_ISSUER} | cut -d "/" -f4)
+endif::azure[]
+----
++
+[NOTE]
+====
+You might need to specify some of the values if the corresponding resource names are not equal to the values derived from the cluster name.
+====
+ifdef::aws[]
+** For {aws-short} clusters that store the OIDC configuration in a public S3 bucket, configure the following environment variables:
++
+[source,text]
+----
+AWS_BUCKET=$(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} | awk -F'://' '{print$2}' |awk -F'.' '{print$1}')
+CLUSTER_NAME=$(echo $AWS_BUCKET|awk -F'-oidc' '{print$1}')
+----
+
+** For {aws-short} clusters that store the OIDC configuration in a private S3 bucket that is accessed by the IAM identity provider through a public CloudFront distribution URL, complete the following steps:
+
+... Determine the public CloudFront distribution URL by running the following command:
++
+[source,terminal]
+----
+$ basename $(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} )
```

Comment:
as mentioned in Slack, wondering why we use `basename` here to wrap the normal `oc get` command

Reply:
Resolved via Slack